### DISCUSSION TITLE
@types/conventional-changelog-writer: add type definition for parseArray method

### DISCUSSION THREAD

**Discussion by MartinMuzatko** [DefinitelyTyped/DefinitelyTyped#54492] [Issues with a @types package]
the function in question is defined here:
https://github.com/conventional-changelog/conventional-changelog/blob/master/packages/conventional-changelog-writer/index.js#L174

Recently, @sandersn, @JasonHK and @armanio123 authored the d.ts file.

I'd be glad to help out adding the type definition. Should be as simple as:

```ts
declare function parseArray<TCommit extends Commit = Commit, TContext extends Context = Context>(rawCommits: TCommit[], context?: Partial<TContext>, options?: Options<TCommit, TContext>): Stream.Transform;
```
not sure however, how to add a function as part of a property of the "default" exported function.

**Comment by sandersn** [comment]
Add that function declaration inside the `declare namespace conventionalChangelog {` block; that's how to add properties to a function in a d.ts.
DT is super picky about modifiers, so I think you'll have to just write `function` not `declare function`.

(@JasonHK is the real maintainer of the types, @armanio123 and I maintain DT as a whole, so we've made automated changes in the past.)

**Reply by MartinMuzatko** [reply]
Thank you for the quick response.
I'd love to prepare a PR (after reading how contributing works here)  :) Thank you for pointing out how it works!
